traffic_ops/traffic_ops_golang/invalidationjobs/invalidationjobs.go

```diff
@@ -203,17 +301,124 @@ func selectMaxLastUpdatedQuery(where string) string {
 	select max(last_updated) as t from last_deleted l where l.table_name='job') as res`
 }
 
+// Deprecated, only to be used with versions below 4.0
+const readQuery = `
+SELECT job.id,
+        'PURGE' AS keyword,
+		ttl_hr,
+		asset_url,
+		start_time,
+		u.username as createdBy,
+		ds.xml_id as dsId
+FROM job
+JOIN tm_user u ON job.job_user = u.id
+JOIN deliveryservice ds ON job.job_deliveryservice = ds.id
+`
+
+// Almost the same as readQuery, but returns appropriate values for API 4.0+
+const readQueryV40 = `
+SELECT job.id,
+	   asset_url,
+	   u.username as createdBy,
+	   ds.xml_id,
+	   ttl_hr,
+	   invalidation_type,
+	   start_time
+FROM job
+JOIN tm_user u ON job.job_user = u.id
+JOIN deliveryservice ds ON job.job_deliveryservice = ds.id
+`
+
+// Used by GET requests to `/jobs`, simply returns a filtered list of
+// content invalidation jobs according to the provided query parameters.
+func (job *InvalidationJobV40) Read(h http.Header, useIMS bool) ([]interface{}, error, error, int, *time.Time) {
+	var maxTime time.Time
+	var runSecond bool
+	queryParamsToSQLCols := map[string]dbhelpers.WhereColumnInfo{
+		"id":               dbhelpers.WhereColumnInfo{Column: "job.id", Checker: api.IsInt},
+		"assetUrl":         dbhelpers.WhereColumnInfo{Column: "asset_url"},
+		"startTime":        dbhelpers.WhereColumnInfo{Column: "start_time"},
+		"userId":           dbhelpers.WhereColumnInfo{Column: "job_user", Checker: api.IsInt},
+		"createdBy":        dbhelpers.WhereColumnInfo{Column: `(SELECT tm_user.username FROM tm_user WHERE tm_user.id=job.job_user)`},
+		"deliveryService":  dbhelpers.WhereColumnInfo{Column: `(SELECT deliveryservice.xml_id FROM deliveryservice WHERE deliveryservice.id=job.job_deliveryservice)`},
+		"dsId":             dbhelpers.WhereColumnInfo{Column: "job.job_deliveryservice", Checker: api.IsInt},
+		"invalidationType": dbhelpers.WhereColumnInfo{Column: "invalidation_type"},
+	}
+
+	where, orderBy, pagination, queryValues, errs := dbhelpers.BuildWhereAndOrderByAndPagination(job.APIInfo().Params, queryParamsToSQLCols)
+	if len(errs) > 0 {
+		return nil, util.JoinErrs(errs), nil, http.StatusBadRequest, nil
+	}
```

Comment:
I completed this after you started this PR, but please ensure that [my changes](https://github.com/apache/trafficcontrol/pull/6154/files#diff-3310237988f25e4cdf0632f9d8be54da09f3ac609ac7d8e8ba4dd74254d36bec) made to the prior API versions is replicated in API v4. We need the ability to filter the query by `cdn` and `maxRevalDurationDays`.